### PR TITLE
tests: test defragmentation in different modes

### PR DESCRIPTION
**Pull request by antmat**
Prettify tests, sync with upcoming PR

### CODE REVIEW

File: tests/unit/defrag.cpp [hunk 1]
```diff
@@ -23,37 +23,46 @@
...
+	return wrapper;
 }
 
-
+template<class D>
 void fill_eblob(eblob_wrapper &wrapper,
-                std::vector<item_t> &shadow_elems,
-                item_generator &generator,
-                size_t total_records) {
+		std::vector<item_t> &shadow_elems,
```
Comment (s-mx) [inline tests/unit/defrag.cpp]: here spaces instead of tabs

File: tests/unit/defrag.cpp [hunk 2]
```diff
@@ -23,37 +23,46 @@
...
+	: items(items_)
+	, expect_blob_sorted(true) {
+	}
+
+	explicit iterator_private(std::vector<item_t> &items_, bool expect_blob_sorted_)
+	: items(items_)
+	, expect_blob_sorted(expect_blob_sorted_) {
 	}
 
 	std::vector<item_t> &items;
```
Comment (s-mx) [inline tests/unit/defrag.cpp:62]: items_ and after the same way

File: tests/unit/eblob_wrapper.h [hunk 3]
```diff
@@ -11,6 +11,24 @@
...
+public:
+	eblob_config_test_wrapper();
+	eblob_config_test_wrapper(const eblob_config_test_wrapper &) = delete;
+	eblob_config_test_wrapper &operator=(const eblob_config_test_wrapper &) = delete;
+	eblob_config_test_wrapper(eblob_config_test_wrapper &&) = default;
+	eblob_config_test_wrapper &operator=(eblob_config_test_wrapper &&) = default;
+	void reset_dirs();
+
+public:
+	eblob_config config;
```
Comment (s-mx) [inline tests/unit/eblob_wrapper.h:24]: config_

File: tests/unit/eblob_wrapper.h [hunk 4]
```diff
@@ -100,35 +65,74 @@ class eblob_wrapper {
...
+template<class D>
 class item_generator {
 public:
 
-	static constexpr uint64_t DEFAULT_RANDOM_SEED = 42;
-
-	explicit item_generator(eblob_wrapper &wrapper, uint64_t seed = DEFAULT_RANDOM_SEED)
-	: wrapper_(wrapper)
-	, gen_(std::mt19937(seed)) {
+	explicit item_generator(eblob_wrapper &wrapper_, D d, uint64_t seed_ = DEFAULT_RANDOM_SEED)
```
Comment (s-mx) [inline tests/unit/eblob_wrapper.h]: doesn't need in explicit here

File: tests/unit/defrag.cpp [hunk 5]
```diff
@@ -67,19 +76,24 @@ int iterate_callback(struct eblob_disk_control *dc,
...
 	BOOST_REQUIRE(!item.checked); //  item already checked
-	BOOST_REQUIRE(dc->data_size == item.value.size());  // sizes mismatch
+	BOOST_REQUIRE_EQUAL(dc->data_size, item.value.size());  // sizes mismatch
 
 	std::vector<char> data(dc->data_size);
 	int ret = __eblob_read_ll(fd, data.data(), dc->data_size, data_offset);
-	BOOST_REQUIRE(ret == 0);  // can't read data
-	BOOST_REQUIRE(data == item.value);  // content of the value differs
+	BOOST_REQUIRE_EQUAL(ret, 0);  // can't read data
+	BOOST_REQUIRE(data == item.value);  // contet of value differ
```
Comment (s-mx) [inline tests/unit/defrag.cpp]: typo

File: tests/unit/defrag.cpp (hunk 2 above)
Comment (antmat) [reply tests/unit/defrag.cpp:62]: it's public

File: tests/unit/eblob_wrapper.h (hunk 3 above)
Comment (antmat) [reply tests/unit/eblob_wrapper.h:24]: also public

File: tests/unit/eblob_wrapper.h (hunk 4 above)
Comment (antmat) [reply tests/unit/eblob_wrapper.h]: thnx

**Pull request merged**